falcon_multipart/middleware.py

```diff
@@ -17,7 +17,10 @@ def process_request(self, req, resp, **kwargs):
         # This must be done to avoid a bug in cgi.FieldStorage.
         req.env.setdefault('QUERY_STRING', '')
 
-        form = self.parse(stream=req.stream, environ=req.env)
+        if hasattr(req.stream, 'stream'):
```

Comment:
I'd do it a bit differently, something like:

    stream = req.stream.stream if hasattr(req.stream, 'stream') else req.stream
    form = self.parse(stream=stream, environ=req.env)

Plus I'd add a small comment to explain why we need this check.